syncer/syncer.go

```diff
@@ -2451,6 +2462,7 @@ func (s *Syncer) trackDDL(usedSchema string, sql string, tableNames [][]*filter.
 			ec.tctx.L().Error("cannot track DDL", zap.String("schema", usedSchema), zap.String("statement", sql), log.WrapStringerField("location", ec.currentLocation), log.ShortError(err))
 			return terror.ErrSchemaTrackerCannotExecDDL.Delegate(err, sql)
 		}
+		s.exprFilterGroup.ResetExprs(srcTable.Schema, srcTable.Name)
```

Comment:
IMO `exprFilterGroup` is reset because the table structure which came from schema tracker has been changed. `Syncer.reset()` doesn't reset schema tracker, so I think it's no need to reset `exprFilterGroup`.